app/views/home/_listing.html.erb

```diff
@@ -0,0 +1,36 @@
+<% @users.each do |user| %>
+  <tr>
+    <% if user.email %>
```

Comment:
That's nice you handle the case where there is no data saved. But handling it in a view introduced an anti-pattern of using a lot of logic in a view. Consider using some sort of decorator / presenter pattern here.